### PR TITLE
[16.0][MIG] sale_line_refund_to_invoice_qty: Migration to 16.0

### PR DESCRIPTION
**Pull request by JavierIniesta**
#1250 
Add fix: #1383

### CODE REVIEW

File: sale_line_refund_to_invoice_qty/models/account.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        if self.env.context.get("sale_qty_to_reinvoice"):
+            res.line_ids.write({"sale_qty_to_reinvoice": True})
+        return res
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    sale_qty_to_reinvoice = fields.Boolean(
+        help="Leave it marked if you will reinvoice the same sale order line",
```
Comment (ljsalvatierra-factorlibre) [inline sale_line_refund_to_invoice_qty/models/account.py:24]: `copy=False`?

File: sale_line_refund_to_invoice_qty/models/sale.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            for invoice_line in line.invoice_lines:
+                if (
+                    invoice_line.move_id.state != "cancel"
+                    and invoice_line.move_id.move_type == "out_refund"
+                    and not invoice_line.sale_qty_to_reinvoice
+                ):
+                    qty_to_invoice -= invoice_line.product_uom_id._compute_quantity(
+                        invoice_line.quantity, line.product_uom
+                    )
+            line.qty_to_invoice = qty_to_invoice
```
Comment (ljsalvatierra-factorlibre) [inline sale_line_refund_to_invoice_qty/models/sale.py]: I would rewrite only if the conditions are met.

File: sale_line_refund_to_invoice_qty/readme/CONTRIBUTORS.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+* Jordi Masvidal <jordi.masvidal@forgeflow.com>
```
Comment (ljsalvatierra-factorlibre) [inline sale_line_refund_to_invoice_qty/readme/CONTRIBUTORS.rst:1]: I think you should be added as a contributor.

**Review by ljsalvatierra-factorlibre — CHANGES_REQUESTED** [review]

File: sale_line_refund_to_invoice_qty/models/account.py (hunk 1 above)
Comment (JavierIniesta) [reply sale_line_refund_to_invoice_qty/models/account.py:24]: Added.

File: sale_line_refund_to_invoice_qty/models/sale.py (hunk 2 above)
Comment (JavierIniesta) [reply sale_line_refund_to_invoice_qty/models/sale.py]: Modified.

File: sale_line_refund_to_invoice_qty/readme/CONTRIBUTORS.rst (hunk 3 above)
Comment (JavierIniesta) [reply sale_line_refund_to_invoice_qty/readme/CONTRIBUTORS.rst:1]: Added.

**Review by ljsalvatierra-factorlibre — APPROVED** [review]

**Review by RodrigoBM — APPROVED** [review]

**Comment by BT-jmichaud** [comment]
@AaronHForgeFlow 
Hello,
Who can merge this PR?
Thank you

**Comment by AaronHForgeFlow** [comment]
@BT-jmichaud this LGTM but there is an open question in this PR: https://github.com/OCA/account-invoicing/pull/1383 that is forward ported here. I prefer to wait until that question is solved or until @RodrigoBM reviews this one.

**Comment by AaronHForgeFlow** [comment]
Well, as long as @RodrigoBM approved this one I think the question is solved.

**Comment by AaronHForgeFlow** [comment]
Well, as long as @RodrigoBM approved this one I think the question is solved.

**Review by AaronHForgeFlow — APPROVED** [review]

**Pull request merged**